volatility3/framework/plugins/windows/driver_signing_enforcement.py

```diff
@@ -0,0 +1,124 @@
+        """
+        for mod in modules.Modules.list_modules(self.context, 
+                                                kernel.layer_name,
+                                                kernel.symbol_table_name):
+     
+            try:
+                module_name = mod.BaseDllName.get_string()
+            except exceptions.InvalidAddressException:
+                continue
+
+            if module_name.lower() == "ci.dll":
+                return mod
+
+        return None
+
+    def _check_win10_64bit(self, kernel):
+        """
+        Checks if the analyzed memory sample is from
+        a Windows 10 64bit system.
+
+        Returns:
+            bool
+        """
+        kuser = info.Info.get_kuser_structure(self.context, 
+                                              kernel.layer_name,
+                                              kernel.symbol_table_name)
+
+        nt_major_version = int(kuser.NtMajorVersion)
+
+        is_64bit = symbols.symbol_table_is_64bit(self.context, kernel.symbol_table_name)
+
+        return nt_major_version >= 6 and is_64bit
+    
+    def _generator(self):
+        """
+        Uses the PDB of CI.DLL to locate g_CiOptions and report its value
+        """
+        kernel = self.context.modules[self.config['kernel']]
+
+        if not self._check_win10_64bit(kernel):
+            vollog.error("This plugin only supports 64-bit Windows 10 or later samples.")
+            return
+
+        ci_mod = self._get_ci_dll(kernel)
+        if ci_mod is None:
+            vollog.error("CI.DLL not found in the kernel module list. Cannot proceed.")
+            return
+
+        # we do not have a fallback mechanism, so if the PDB cannot load then we are done
+        try:
+            ci_symbols = pdbutil.PDBUtility.symbol_table_from_pdb(self.context,
+                                                                  interfaces.configuration.path_join(
+                                                                      self.config_path, 'ci'),
+                                                                  kernel.layer_name,
+                                                                  "ci.pdb",
+                                                                  ci_mod.DllBase,
+                                                                  ci_mod.SizeOfImage)
+
+        except exceptions.VolatilityException as e:
+            vollog.error("Unable to analyze the PDB file for CI.DLL: {}.".format(str(e)))
+            return
+
+        ci_module = self.context.module(ci_symbols, layer_name = kernel.layer_name, offset = ci_mod.DllBase)
+
+        options_address = ci_module.get_absolute_symbol_address("g_CiOptions")
+      
+        options = kernel.object(object_type = "unsigned int", offset = options_address, absolute = True)
+
+        yield (0, (options != 0, options))
+
+    def run(self):
+        return renderers.TreeGrid([("Driver Signing Enforcement", bool), ("g_CiOptions Value", int)],
```

Comment:
This feels like it will only return a single row ever, are there other options in the structure it would be useful to return?  Seems a waste to do all the work to pull out this one location, and not stop and see if there's any other global flags/variables we could tell people about?